dunwoody_disaster/views/MapScreen.py

```diff
@@ -135,17 +135,38 @@ def from_json(json: dict, char: Character) -> "MapScreen":
 
         return map
 
-    @staticmethod
-    def build_map(char: Character) -> "MapScreen":
-        test_enemy = CharacterFactory.createTestChar()
-        test_enemy.name = "test enemy"
-        ms = MapScreen(char, None)
-        ms.setAsset("MainMap")
-        ms.addRoom("Bus Stop", (419, 700), test_enemy, "no_texture")
-        ms.addRoom("Court Yard", (693, 559), test_enemy, "no_texture")
-        ms.addRoom("Commons", (451, 449), test_enemy, "no_texture")
-        ms.addRoom("Math", (236, 359), test_enemy, "no_texture")
-        ms.addRoom("English", (770, 366), test_enemy, "no_texture")
-        ms.addRoom("Science", (490, 217), test_enemy, "no_texture")
-        ms.addRoom("Dean's Office", (90, 589), test_enemy, "no_texture")
-        return ms
+
+class MapScreen(QWidget):
+    def __init__(self, map: Map):
+        super().__init__()
+        self.map = map
+        self._callback = DD.unimplemented
+
+        self.init_ui()
+
+    def onEnter(self, callback: Callable):
+        """
+        :param callback: A callback function that takes the room info as a parameter
+        This callback will be triggered when the user selects which room to enter
+        """
+        self._callback = callback
+
+    def keyPressEvent(self, event: QKeyEvent):
```

Comment:
```suggestion
    def KeyPressEvent(self, event: QKeyEvent):
```

Reply:
This method is being over-ridden, so we can't change the casing.